### ISSUE TITLE
[BUG] Criteria depending on `ReturnCriterion` includes the base

### ISSUE THREAD

**Issue by nimo23** [ta4j/ta4j#1076]
**Describe the bug**
I don't know exactly if this is a bug, but we should discuss it to make sure that it's correct.

The following critera:
- `ReturnOverMaxDrawdownCriterion`
- `AverageReturnPerBarCriterion` 
- `LinearTransactionCostCriterion`

use the `ReturnCriterion` which **includes the base** for their calculations. Is this correct? Or should we **exclude the base** by doing:

### (1) AverageReturnPerBarCriterion:

```
public class AverageReturnPerBarCriterion extends AbstractAnalysisCriterion {

   private final ReturnCriterion grossReturn = new ReturnCriterion(false);
..
```

### (2) ReturnOverMaxDrawdownCriterion:

```
public class ReturnOverMaxDrawdownCriterion extends AbstractAnalysisCriterion {

   private final AnalysisCriterion grossReturnCriterion = new ReturnCriterion(false);
..
```

### (3) LinearTransactionCostCriterion:

```
public class LinearTransactionCostCriterion extends AbstractAnalysisCriterion {

public LinearTransactionCostCriterion(double initialAmount, double a, double b) {
        this.initialAmount = initialAmount;
        this.a = a;
        this.b = b;
        grossReturn = new ReturnCriterion(false);
}
..
}
```

What do you think?

### For (1) AverageReturnPerBarCriterion:

The general formula for `ReturnOverMaxDrawdownCriterion`:

`Return over Max. Drawdown = Total Return / Max. Drawdown`

or:

`Return over Max. Drawdown = (Total Return - Max. Drawdown) / Max. Drawdown`

with the general formula for `TotalReturn`:

`Total Return = (Final Investment Value - Initial Investment Value) / Initial Investment Value`

If we have a `MaxDrawdownCriterion` = 1 (= total loss) and  `ReturnCriterion` = 1 (because we include the base), then we have `ReturnOverMaxDrawdownCriterion` = 1. I don't know if this makes sense at all. I think, the "Total Return in percentage" should not include the base value of 100% Am I right? If so, then we should exclude the base from the calculation of `ReturnOverMaxDrawdownCriterion`.

**Comment by phong-phuong** [comment]
I agree. For most metrics where return is used, it is the **excess return** (without the base) that is of interest.
Otherwise, for a strategy that does absolutely nothing, **100% return** per bar looks amazing, but in fact doesn't generate any yield.
